Feat/feature screen one domain

anarahari opened the pull request.

anarahari committed:
feat : Feature screen one, domain module setup

anarahari committed:
feat : Feature screen one, domain module setup

anarahari committed:
Merge branch 'feat/Workflow_EmptyModules' into feat/Feature_Screen_One_Domain

# Conflicts:
#	common/build.gradle.kts

anarahari committed:
feat : use case, repository, pagination setup

anarahari committed:
feat : use case, repository, pagination setup

anarahari committed:
Merge branch 'feat/Workflow_EmptyModules' into feat/Feature_Screen_One_Domain

anarahari committed:
feat : setup data layer for screen one

anarahari committed:
feat : removed dispatcher in yaml file

anarahari committed:
feat : changed sequence of lint check in yaml file

anarahari committed:
feat : removed spaces for ktlintcheck

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

# Conflicts:
#	domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt
#	domain/src/main/java/com/compose/domain/repository/CharacterRepository.kt
#	domain/src/main/java/com/compose/domain/usecase/GetCharactersUseCase.kt

anarahari committed:
feat : screen one data module setup with graphql, repo impl, dagger modules

shreyansh-mm commented on .idea/compiler.xml:
This file should not be committed to VCS

anarahari replied:
done

shreyansh-mm commented on .idea/gradle.xml:
This file should not be committed to VCS

anarahari replied:
done

shreyansh-mm commented on .idea/graphql-settings.xml:
This file should not be committed to VCS

anarahari replied:
done

shreyansh-mm commented on .idea/misc.xml:
This file should not be committed to VCS

anarahari replied:
done

shreyansh-mm commented on domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt:
```diff
@@ -0,0 +1,46 @@
+import com.apollographql.apollo.api.Optional
+import com.common.graphql.GetCharactersQuery
+
+class CharacterPagingSource(
```
This should not be in domain

anarahari replied:
done
moved it to data module

shreyansh-mm commented on domain/src/main/java/com/compose/domain/repository/CharacterRepository.kt:
```diff
@@ -0,0 +1,10 @@
+
+interface CharacterRepository {
+
+    suspend fun getCharacters(): Flow<PagingData<GetCharactersQuery.Result>>
```
we should not be returning `GetCharactersQuery.Result`, we need to create our own model and map the response to our class using mapper/extension functions

anarahari committed:
feat : Pr review changes

anarahari committed:
feat : gitignore changes

anarahari committed:
feat : gitignore changes

anarahari committed:
.idea removed and changes to .gitignore

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
feat : Added datamapper, repositoryimpl changes to integrate data mapper for character model

anarahari committed:
feat : ktlintcheck changes for spaces

anarahari committed:
feat : CommonConstants marked as internal

shreyansh-mm commented on domain/src/main/java/com/compose/domain/usecase/GetCharactersUseCase.kt:
```diff
@@ -0,0 +1,13 @@
+
+class GetCharactersUseCase @Inject constructor(private val characterRepository: CharacterRepository) {
+
+    suspend fun getCharacters(): Flow<List<CharacterModel>> {
```
Methods of usecase are always called `invoke`

shreyansh-mm commented on domain/src/main/java/com/compose/domain/mapper/CharacterModel.kt:
```diff
@@ -0,0 +1,10 @@
+package com.compose.domain.mapper
+
+data class CharacterModel(
+    var id: String,
+    var name: String,
+    var image: String,
+    var gender: String,
+    var species: String,
+    var status: String
```
Should be `val` not `var`

anarahari committed:
feat : domain

- Character model changed to val
- use case function name changed to invoke

anarahari committed:
feat : data

- CharacterPagingSource removed commented code

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
feat : domain

- Changed the return type of getCharacters to List<T>

anarahari committed:
feat : domain

- Changed the return type of getCharacters to List<T>

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
feat : domain

- Changed the return type of getCharacters to List<T> in usecase

anarahari committed:
Merge branch 'feat/Feature_Screen_One_Domain' into feat/Feature_Screen_One_Data

anarahari committed:
feat : data

- Changed the implementation type of getCharacters to   List<T> in repository
- Nullable return of charactermodel in datamapper

anarahari committed:
Merge pull request #3 from anarahari/feat/Feature_Screen_One_Data

Feat/feature screenone data module
